File: backend/app/admin/service/internal/service/notification_message_category.go
package service

import (
	"context"

	"github.com/go-kratos/kratos/v2/log"
	"github.com/tx7do/go-utils/trans"
	pagination "github.com/tx7do/kratos-bootstrap/api/gen/go/pagination/v1"
	"google.golang.org/protobuf/types/known/emptypb"

	"kratos-admin/app/admin/service/internal/data"

	adminV1 "kratos-admin/api/gen/go/admin/service/v1"
	internalMessageV1 "kratos-admin/api/gen/go/internal_message/service/v1"

	"kratos-admin/pkg/middleware/auth"
)

type NotificationMessageCategoryService struct {
	adminV1.NotificationMessageCategoryServiceHTTPServer

	log *log.Helper

	uc *data.NotificationMessageCategoryRepo
}

func NewNotificationMessageCategoryService(uc *data.NotificationMessageCategoryRepo, logger log.Logger) *NotificationMessageCategoryService {
	l := log.NewHelper(log.With(logger, "module", "notification-message-category/service/admin-service"))
	return &NotificationMessageCategoryService{
		log: l,
		uc:  uc,
	}
}

func (s *NotificationMessageCategoryService) List(ctx context.Context, req *pagination.PagingRequest) (*internalMessageV1.ListNotificationMessageCategoryResponse, error) {
	return s.uc.List(ctx, req)
}

func (s *NotificationMessageCategoryService) Get(ctx context.Context, req *internalMessageV1.GetNotificationMessageCategoryRequest) (*internalMessageV1.NotificationMessageCategory, error) {
	return s.uc.Get(ctx, req)
}

func (s *NotificationMessageCategoryService) Create(ctx context.Context, req *internalMessageV1.CreateNotificationMessageCategoryRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("invalid parameter")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.CreateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Create(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *NotificationMessageCategoryService) Update(ctx context.Context, req *internalMessageV1.UpdateNotificationMessageCategoryRequest) (*emptypb.Empty, error) {
	if req.Data == nil {
		return nil, adminV1.ErrorBadRequest("invalid parameter")
	}

	// 获取操作人信息
	operator, err := auth.FromContext(ctx)
	if err != nil {
		return &emptypb.Empty{}, err
	}

	req.Data.UpdateBy = trans.Ptr(operator.UserId)

	if err = s.uc.Update(ctx, req); err != nil {
		return nil, err
	}

	return &emptypb.Empty{}, nil
}

func (s *NotificationMessageCategoryService) Delete(ctx context.Context, req *internalMessageV1.DeleteNotificationMessageCategoryRequest) (*emptypb.Empty, error) {
	if err := s.uc.Delete(ctx, req); err != nil {
		return nil, err
	}
	return &emptypb.Empty{}, nil
}

File: frontend/apps/admin/src/rpc/service/notification_message_category.ts
import type { NotificationMessageCategoryService } from '#/rpc/api/admin/service/v1/i_notification_message_category.pb';
import type { Empty } from '#/rpc/api/google/protobuf/empty.pb';
import type {
  CreateNotificationMessageCategoryRequest,
  DeleteNotificationMessageCategoryRequest,
  GetNotificationMessageCategoryRequest,
  ListNotificationMessageCategoryResponse,
  NotificationMessageCategory,
  UpdateNotificationMessageCategoryRequest,
} from '#/rpc/api/internal_message/service/v1/notification_message_category.pb';
import type { PagingRequest } from '#/rpc/api/pagination/v1/pagination.pb';

import { requestClient } from '#/rpc/request';

/** 通知消息分类管理服务 */
class NotificationMessageCategoryServiceImpl
  implements NotificationMessageCategoryService
{
  async Create(
    request: CreateNotificationMessageCategoryRequest,
  ): Promise<Empty> {
    return await requestClient.post<Empty>(
      '/notifications:categories',
      request,
    );
  }

  async Delete(
    request: DeleteNotificationMessageCategoryRequest,
  ): Promise<Empty> {
    return await requestClient.delete<Empty>(
      `/notifications:categories/${request.id}`,
    );
  }

  async Get(
    request: GetNotificationMessageCategoryRequest,
  ): Promise<NotificationMessageCategory> {
    return await requestClient.get<NotificationMessageCategory>(
      `/notifications:categories/${request.id}`,
    );
  }

  async List(
    request: PagingRequest,
  ): Promise<ListNotificationMessageCategoryResponse> {
    return await requestClient.get<ListNotificationMessageCategoryResponse>(
      '/notifications:categories',
      {
        params: request,
      },
    );
  }

  async Update(
    request: UpdateNotificationMessageCategoryRequest,
  ): Promise<Empty> {
    const id = request.data?.id;

    console.log('UpdateNotificationMessageCategory', request.data);
    if (request.data !== null && request.data !== undefined) {
      request.data.id = undefined;
    }
    return await requestClient.put<Empty>(
      `/notifications:categories/${id}`,
      request,
    );
  }
}

export const defNotificationMessageCategoryService =
  new NotificationMessageCategoryServiceImpl();
